pandas/io/pytables.py

```diff
@@ -3867,6 +3877,9 @@ def get_blk_items(mgr, blocks):
             if data_columns and len(b_items) == 1 and b_items[0] in data_columns:
                 klass = DataIndexableCol
                 name = b_items[0]
+                if not (name is None or isinstance(name, str)):
+                    # TODO: should the message here be more specifically non-str?
```

Comment:
is this actually hit anywhere? I think this is by-definition not possible.

Reply:
It is.  we have a test that gets here with an np.int64